### PR TITLE
Fixes/sma 187 explicit tests per zellic audit

### PR DESCRIPTION
**Pull request by filmakarov**
# Summary

Added explicit test cases per Zellic Audit of Batched Session Router

Related Issue: # (issue number)

## Change Type

- [ ] Bug Fix
- [ ] Refactor
- [ ] New Feature
- [ ] Breaking Change
- [ ] Documentation Update
- [ ] Performance Improvement
- [x] Other

# Checklist

- [x] My code follows this project's style guidelines
- [x] I've reviewed my own code
- [x] I've added comments for any hard-to-understand areas
- [x] I've updated the documentation if necessary
- [x] My changes generate no new warnings
- [x] I've added tests that prove my fix is effective or my feature works
- [x] All unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published

### CODE REVIEW

**Comment by livingrockrises** [comment]
Should we change the base to develop and back merge all latest code from develop

**Comment by filmakarov** [comment]
Base changed. 
Zellic confirmed all is good, need to merge and pass it to Kawach

File: test/module/BatchedSessionRouter.Module.specs.ts [hunk 1]
```diff
@@ -943,4 +937,309 @@ describe("SessionKey: Batched Session Router", async () => {
...
+      const approveCallData = IERC20.interface.encodeFunctionData("approve", [
+        mockProtocol.address,
+        tokenAmountToTransfer,
+      ]);
+      const interactCallData = MockProtocol.interface.encodeFunctionData(
+        "interact",
+        [mockToken.address, tokenAmountToTransfer]
+      );
+
+      /*
```
Comment (ankurdubey521) [inline test/module/BatchedSessionRouter.Module.specs.ts:1137]: do we need this block of commented out code?

**Review by ankurdubey521 — APPROVED** [review]
LGTM

**Review by AmanRaj1608 — APPROVED** [review]

**Pull request merged**